### PR TITLE
Switching sensors on and off

### PR DESCRIPTION
**Pull request by IsabelleTornqvist**

### CODE REVIEW

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 1]
```diff
@@ -18,17 +22,27 @@
...
 boolean obstacle;
 boolean completeCommand;
 boolean flame;
+//For turning modules on and off
+boolean flameActivation = true;
+boolean temperatureActivation = true;
+boolean ultrasonicActivation = true; 
+boolean servoActivation = true;     
+boolean motorActivation = true;
+boolean RIP = true;
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Needs a comment to specify what this boolean is for

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 2]
```diff
@@ -1,6 +1,10 @@
-#include <NewPing.h>
+#include <NewPing.h> // https://bitbucket.org/teckel12/arduino-new-ping/downloads/
 #include <Servo.h>
+#include <OneWire.h> //https://github.com/PaulStoffregen/OneWire
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Needs to be specified what this library does

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+#include <DallasTemperature.h> //https://github.com/milesburton/Arduino-Temperature-Control-Library
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Needs to be specified what this library does

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 1 above)
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: All the activations use the suffix Activation but not this one.

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 4]
```diff
@@ -57,16 +79,49 @@ void setup() {
     pinMode(FLAME_SENSOR_PIN, INPUT);
     
     Serial.begin(9600);  
+
+  // Start the sensor
+    sensors.begin();    
+    if (!sensors.getAddress(thermometer, 0)) Serial.println("Unable to find address for Device 0");
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Why is there a serial.print here

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 5]
```diff
@@ -57,16 +79,49 @@ void setup() {
     pinMode(FLAME_SENSOR_PIN, INPUT);
     
     Serial.begin(9600);  
+
+  // Start the sensor
+    sensors.begin();
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: What is this used for?

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 4 above)
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Why are you printing to the serial? Doesn't that get passed to the android client and cause issues?

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 6]
```diff
@@ -57,16 +79,49 @@ void setup() {
...
+
+//Added by Isabelle Törnqvist 2017-04-18
+// function to print the temperature for a device
+void printTemperature(DeviceAddress deviceAddress)
+{  
+  oldTemp = tempC;
+  tempC = (int)sensors.getTempC(deviceAddress);
+  String tempString;
+
+//Check temperature
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Doesn't this convert the temperature to a string for sending it? not check it.

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 7]
```diff
@@ -18,17 +22,27 @@
 int flame_pin;
 int velocity;
 int range;
+int tempC;
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Why is the temp variables declared next to the range and velocity? Are they related?

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 8]
```diff
@@ -136,6 +226,12 @@ void loop() {
             motor.write(START_POSITION_MOTOR_SERVO);
         }
     }
+
+  if(temperatureActivation){
+    //for temperature sensor
+    sensors.requestTemperatures(); // Get temperatures
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Needs a delay

**Review by timjon — CHANGES_REQUESTED** [review]

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 9]
```diff
@@ -76,7 +131,37 @@ void loop() {
                 motor.write(velocity);
             }
             break;
-     
+
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: At least make a comment for all of these cases.

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 1 above)
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Full capital variable decleration. Please don't do this Tim.

File: SmartCar/Arduino/sketch_main/sketch_main.ino (hunk 7 above)
Comment (IsabelleTornqvist) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: yes

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 10]
```diff
@@ -114,10 +207,11 @@ void loop() {
         Serial.print(range_string);
         lastDistance = range;
         distance_Delay = 0;
-    }
+    }   
+  }
 
     //Allows the car to drive backwards if an obstacle is in from of the car
-    if (velocity < 90){
+    if (velocity < 90 && !motorActivation){
```
Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Shouldnt be not motorActivation

Comment (timjon) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Shouldnt be a not motorActivation since then it wont do this while the motor is on

**Review by timjon — CHANGES_REQUESTED** [review]

File: SmartCar/Arduino/sketch_main/sketch_main.ino [hunk 11]
```diff
@@ -87,25 +176,29 @@ void loop() {
         completeCommand = false;
     }
```
Comment (pontuslae) [inline SmartCar/Arduino/sketch_main/sketch_main.ino]: Add a few line breaks here to seperate flame stuff from other stuff. So it's easier to locate it.

**Review by timjon — APPROVED** [review]

**Review by pontuslae — APPROVED** [review]
All my concerns have been adressed.

**Review by ElaineQvarnstrom — APPROVED** [review]

**Pull request merged**